CRM/Mailing/BAO/Mailing.php

```diff
@@ -3118,4 +3136,14 @@ public static function getTemplateTypeNames() {
     return $r;
   }
 
+  /**
+   * Get a safe temporary filename for MySQL SELECT INTO / LOAD DATA statements 
+   *
+   * @return string
+   */
+  private static function tmpfile_for_select_into($tmpDirectory, $mailingID) {
```

Comment:
I tried using tempnam but it checks the local machine to see if it can write to the directory and falls back to /tmp if it's not there or not-writable.  That obviously won't work - we aren't writing on the webserver anyway.

``` Creates a file with a unique filename, with access permission set to 0600, in the specified directory. If the directory does not exist or is not writable, tempnam() may generate a file in the system's temporary directory, and return the full path to that file, including its name. ```

https://www.php.net/manual/en/function.tempnam.php